algorithms_task_2/student/include/impl.h

```diff
@@ -9,4 +9,9 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix)
+{
+     return source.rfind(prefix, source.find_first_not_of(' ')) == 0; // ignore initial whitespaces
```

Comment:
I'd use equal here for bool, and mismatch for iterators.
Find will not stop if case if char mismatch